src/main/java/uk/gov/hmcts/reform/bulkscanprocessor/tasks/BlobProcessorTask.java

```diff
@@ -93,114 +87,112 @@ public ServiceBusHelper serviceBusHelper() {
+            .blockingGet();
 
-        if (!isReadyToBeProcessed(cloudBlockBlob)) {
+        if (!isReadyToBeProcessed(properties.headers().lastModified())) {
             return;
         }
 
         log.info("Processing zip file {}", zipFilename);
 
         Envelope existingEnvelope =
-            envelopeProcessor.getEnvelopeByFileAndContainer(container.getName(), zipFilename);
+            envelopeProcessor.getEnvelopeByFileAndContainer(containerName, zipFilename);
         if (existingEnvelope != null) {
-            deleteIfProcessed(cloudBlockBlob, existingEnvelope);
+            deleteIfProcessed(blockBlobURL, existingEnvelope);
             return;
         }
 
-        CloudBlockBlob blobWithLeaseAcquired = acquireLease(cloudBlockBlob, container.getName(), zipFilename);
+        LeaseStateType leaseStateType = properties.headers().leaseState();
+        boolean leaseAvailable = azureStorageHelper.checkLeaseAvailable(containerName, zipFilename, leaseStateType);
 
-        if (Objects.nonNull(blobWithLeaseAcquired)) {
-            BlobInputStream blobInputStream = blobWithLeaseAcquired.openInputStream();
+        if (leaseAvailable) {
+            byte[] blob = azureStorageHelper.downloadBlob(blockBlobURL).blockingGet().array();
 
             // Zip file will include metadata.json and collection of pdf documents
-            try (ZipInputStream zis = new ZipInputStream(blobInputStream)) {
-                ZipFileProcessor zipFileProcessor = processZipFileContent(zis, zipFilename, container.getName());
+            try (ZipInputStream zis = new ZipInputStream(new ByteArrayInputStream(blob))) {
+                ZipFileProcessor zipFileProcessor = processZipFileContent(zis, zipFilename, containerName);
 
                 if (zipFileProcessor != null) {
                     processParsedEnvelopeDocuments(
                         zipFileProcessor.getEnvelope(),
                         zipFileProcessor.getPdfs(),
-                        blobWithLeaseAcquired,
+                        blockBlobURL,
                         serviceBusHelper
                     );
                 }
             }
         }
     }
 
-    private void deleteIfProcessed(CloudBlockBlob cloudBlockBlob, Envelope envelope) {
+    private void deleteIfProcessed(BlockBlobURL blockBlobURL, Envelope envelope) {
         try {
-            if (cloudBlockBlob != null && envelope.getStatus().isProcessed()) {
-                boolean deleted;
-                if (cloudBlockBlob.exists()) {
-                    deleted = cloudBlockBlob.deleteIfExists();
-                } else {
-                    deleted = true;
-                }
-                if (deleted) {
-                    envelope.setZipDeleted(true);
-                    envelopeProcessor.saveEnvelope(envelope);
-                }
+            if (envelope.getStatus().isProcessed()) {
+                blockBlobURL.delete(DeleteSnapshotsOptionType.INCLUDE,
```

Comment:
extract the delete operation into a method ... simplify.

We need to refactor this to use Rx a bit better ... its not going to be very fast or nice using blocking get everywhere.